test/tls_integration_test.coffee

```diff
@@ -0,0 +1,364 @@
+_ = require 'underscore'
+async = require 'async'
+child_process = require 'child_process'
+{readFileSync} = require 'fs'
+request = require 'request'
+
+should = require 'should'
+temp = require('temp').track()
+
+nsq = require '../src/nsq'
+
+TCP_PORT = 14150
+HTTPS_PORT = 14152
+
+CLIENT_CERT = readFileSync './test/clientcert.cer'
+CLIENT_KEY = readFileSync './test/clientkey.pem'
+CA_CERT = readFileSync './test/cert.pem'
+
+startNSQD = (dataPath, additionalOptions, callback) ->
+  additionalOptions or= {}
+  options =
+    'https-address': "127.0.0.1:#{HTTPS_PORT}"
+    'tcp-address': "127.0.0.1:#{TCP_PORT}"
+    'data-path': dataPath
+    'tls-cert': './test/cert.pem'
+    'tls-key': './test/key.pem'
+    'tls-required': 'true'
+    'tls-client-auth-policy': 'require-verify'
+    'tls-root-ca-file': './test/cert.pem'
+
+  _.extend options, additionalOptions
+  options = _.flatten (["-#{key}", value] for key, value of options)
+  process = child_process.spawn 'nsqd', options.concat(additionalOptions),
+    stdio: ['ignore', 'ignore', 'ignore']
+
+  # Give nsqd a chance to startup successfully.
+  setTimeout _.partial(callback, null, process), 500
+
+topicOp = (op, topic, callback) ->
+  options =
+    method: 'POST'
+    uri: "https://127.0.0.1:#{HTTPS_PORT}/#{op}"
+    qs:
+      topic: topic
+    agentOptions:
+      ca: CA_CERT
+      key: CLIENT_KEY
+      cert: CLIENT_CERT
+      rejectUnauthorized: no
+
+  request options, (err, res, body) ->
+    callback err
+
+createTopic = _.partial topicOp, 'create_topic'
+deleteTopic = _.partial topicOp, 'delete_topic'
+
+# Publish a single message via HTTP
+publish = (topic, message, done) ->
+  options =
+    uri: "https://127.0.0.1:#{HTTPS_PORT}/pub"
+    method: 'POST'
+    qs:
+      topic: topic
+    body: message
+    agentOptions:
+      ca: CA_CERT
+      key: CLIENT_KEY
+      cert: CLIENT_CERT
+      rejectUnauthorized: no
+
+  request options, (err, res, body) ->
+    done? err
+
+describe 'TLS client verification', ->
+  describe 'integration', ->
+    nsqdProcess = null
+    reader = null
+
+    before (done) ->
+      temp.mkdir '/nsq', (err, dirPath) ->
+        startNSQD dirPath, {}, (err, process) ->
+          nsqdProcess = process
+          done err
+
+    after (done) ->
+      nsqdProcess.kill()
+      # Give nsqd a chance to exit before it's data directory will be cleaned up.
+      setTimeout done, 500
+
+    beforeEach (done) ->
+      createTopic 'test', done
+
+    afterEach (done) ->
+      reader.close()
+      deleteTopic 'test', done
+
+    describe 'stream compression and encryption', ->
+      optionPermutations = [
+        { deflate: true }
```

Comment:
This would be fixed by simply adding a keyword `do` :

```coffee
for elem, i in [1, 2, 3, 4, 5]
  do (elem, i) -> 
    it "should be the #{i}th elem", -> 
      elem.should.equal i + 1
```

output:
```
  ✓ should be the 0th elem
  ✓ should be the 1th elem
  ✓ should be the 2th elem
  ✓ should be the 3th elem
  ✓ should be the 4th elem

  5 passing (8ms)
```